pkg/karpenter/providers/helm.go

```diff
@@ -0,0 +1,31 @@
+package providers
+
+import (
+	"bytes"
+	"context"
+
+	"helm.sh/helm/v3/pkg/getter"
+)
+
+// Getter is an interface to support GET to the specified URL.
+//go:generate go run github.com/maxbrunsfeld/counterfeiter/v6 -generate
+//counterfeiter:generate -o fakes/fake_helm_getter.go . Getter
+type Getter interface {
+	// Get file content by url string
+	Get(url string, options ...getter.Option) (*bytes.Buffer, error)
+}
+
+// HelmInstaller deals with setting up Helm related resources.
+//go:generate go run github.com/maxbrunsfeld/counterfeiter/v6 -generate
+//counterfeiter:generate -o fakes/fake_helm_installer.go . HelmInstaller
+type HelmInstaller interface {
+	// AddRepo adds a repository to helm repositories.
+	AddRepo(repoURL string, release string) error
+	// RemoveRepo removes a repository from the list of repositories.
+	RemoveRepo(repoName string) error
+	// InstallChart takes a releaseName's name and a chart name and installs it. If namespace is not empty
+	// it will install into that namespace and create the namespace. Version is required.
+	InstallChart(ctx context.Context, releaseName string, chartName string, namespace string, version string, values map[string]interface{}) error
+	// UninstallChart removes an installed chart from the cluster.
```

Comment:
I think these should be two separate things because each can be repeated potentially. The repo passes, you only have to repeat the chart installation. Also, they do vastly different things code wise.

Also, yes, testing them separately makes for a lot more readable tests and functionality without a lot of jumbled stuff. :)

Reply:
> I think these should be two separate things because each can be repeated potentially. The repo passes, you only have to repeat the chart installation. Also, they do vastly different things code wise.

but do we ever actually do that?

> Also, yes, testing them separately makes for a lot more readable tests and functionality without a lot of jumbled stuff. :)

Thats good enough for me :smile: